phoenix-core/src/main/java/org/apache/phoenix/parse/IndexExpressionParseNodeRewriter.java

```diff
@@ -51,6 +51,12 @@ public Void visit(ColumnParseNode node) throws SQLException {
             return null;
         }
         
+        @Override
+        public boolean visitEnter(FunctionParseNode node) throws SQLException {
+            isParseNodeCaseSensitive = true;
```

Comment:
I don't think there is a downside, in fact I think this is the correct behavior because when we create the index table we set we use a case sensitive version of the expressions string, so I think we always have to include the double quotes

String name = expressionStr.replaceAll("\"", "'");
                        colName = ColumnName.caseSensitiveColumnName(IndexUtil.getIndexColumnName(null, name));